airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -95,42 +99,64 @@ def parse_response(self, response: SearchPager) -> Iterable[Mapping]:
             yield self.google_ads_client.parse_single_result(self.get_json_schema(), result)
 
     def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
-        for customer_id in self.google_ads_client.customer_ids:
-            self._customer_id = customer_id
-            yield {}
+        for customer in self.customers:
+            yield {"customer_id": customer.id}
 
     def read_records(self, sync_mode, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
-        if not stream_slice:
+        if stream_slice is None:
             return []
-        account_responses = self.google_ads_client.send_request(self.get_query(stream_slice), customer_id=self._customer_id)
-        for response in account_responses:
-            yield from self.parse_response(response)
-
 
-class IncrementalGoogleAdsStream(GoogleAdsStream, ABC):
+        customer_id = stream_slice["customer_id"]
+        response_records = self.google_ads_client.send_request(self.get_query(stream_slice), customer_id=customer_id)
+        try:
+            for response in response_records:
+                yield from self.parse_response(response)
+        except GoogleAdsException as exc:
+            if not self.CATCH_API_ERRORS:
+                raise
+            for error in exc.failure.errors:
+                if error.error_code.authentication_error or error.error_code.authorization_error or error.error_code.query_error:
```

Comment:
why not check for the specific issue of an account being inactive instead of ignoring all authn/authz & query errors? (why are we ignoring query errors btw?) 
it seems this way would cause us to ignore actual problems

Reply:
Agree. Fixed